docker-java-core/src/main/java/com/github/dockerjava/core/DockerHttpClient.java

```diff
@@ -0,0 +1,80 @@
+package com.github.dockerjava.core;
+
+import org.immutables.value.Value;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.Closeable;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+
+public interface DockerHttpClient extends Closeable {
+
+    Response execute(Request request);
+
+    interface Response extends Closeable {
+
+        int getStatusCode();
```

Comment:
Does int make sense here? No enum? This has to do with whay you explained in the call, fear of strict typesystem breaking in case of Docker introducing changes? Can we expect this for status code?

Reply:
This is HTTP status code, it is strictly defined and I don't think it is a good idea to use enum here (given all ~500 values it should have) :)